PsfShimMonitor/ColumnSelector.xaml.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+
+namespace TraceShimMonitor
+{
+    /// <summary>
+    /// Interaction logic for ColumnSelector.xaml
+    /// </summary>
+    public partial class ColumnSelector : Window
+    {
+        private DataGrid _SourceDG;
+        private List<CheckBox> _CbList = new List<CheckBox>();
+        
+        public ColumnSelector(DataGrid sdg)
+        {
+            InitializeComponent();
+
+            if (sdg != null)
+            {
+                _SourceDG = sdg;
+                foreach (DataGridColumn dgc in _SourceDG.Columns)
+                {
+                    CheckBox cb = new CheckBox();
+                    cb.Content = dgc.Header;
+                    cb.IsThreeState = false;
+                    cb.IsChecked = (dgc.Visibility == Visibility.Visible);
+                    cb.Checked += Cb_Event_Checked;
+                    cb.Unchecked += Cb_Event_UnChecked;
+                    cb.FontSize = 12;
+                    cb.Height = 20;
+                    _CbList.Add(cb);
+                }
+                ItemsList.ItemsSource = _CbList;
+                //ItemsList.UpdateLayout();
+                //this.Height = 35 + _CbList.Count * 25;  // temp
+                this.Loaded += WhenLoaded;
+            }
+        }
+        
+        private void WhenLoaded(object sender, RoutedEventArgs e)
+        {
+            this.Height = 35 + ItemsList.ActualHeight;           
+        }
+        private void Cb_Event_Checked(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                CheckBox cb = sender as CheckBox;
+                foreach (DataGridColumn dgc in _SourceDG.Columns)
+                {
+                    if (dgc.Header as string == cb.Content as string)
+                    {
+                        dgc.Visibility = Visibility.Visible;
+                    }
+                }
+            }
+            catch
```

Comment:
why the blanket catches?  generally discouraged

Reply:
A testing tool, used to help identify broken programs, itself should be designed to avoid crashing at all costs. If the user clicks on a standard UI function and (for some unplanned reason) the underlying libraries used decide to throw an exception, the code should try to handle it.  In this case, I believe the most appropriate action is to catch and release and hope that the user can still use the displayed data.

Reply:
Please remove all blank catches.  I like the idea of using it as testing, but we should not be hiding bugs and errors from the user.  Additionally, if the code throws an error, that means there could potentially be corrupted data on the UI and we don't want users to be using corrupted data.

Reply:
While my experience strongly leads me to disagree, I have addressed all blank catches. Some are appropriate and are so commented.  When possible, others will result in notification to the user who will be given a choice to terminate or continue on.  When not possible, the condition is rethrown to cause the app to crash.  It is a bad practice to crash a test application that is not used in production when it could continue to provide valuable input to the test person, but we'll just crash anyway.

Reply:
I see where you are going with this.  You believe that hiding the error and letting the information stay on the screen is more valuable than having the application crash, **because**, this is still a test application and not running in production.

If this app were running in production, would the idea be to make sure to crash the system if something went wrong?

Reply:
Is that right, or did I miss something?